src/Reflection/ReflectionMethod.php

```diff
@@ -336,16 +339,24 @@ public function isStatic(): bool
      */
     public function isConstructor(): bool
     {
-        if (strtolower($this->getName()) === '__construct') {
+        return $this->isConstructor;
+    }
+
+    private function computeIsConstructor(string $name, ReflectionClass $declaringClass): bool
+    {
+        if (strtolower($name) === '__construct') {
             return true;
         }
 
-        $declaringClass = $this->getDeclaringClass();
         if ($declaringClass->inNamespace()) {
             return false;
         }
 
-        return strtolower($this->getName()) === strtolower($declaringClass->getShortName());
+        if ($declaringClass->isAnonymous()) {
+            return false;
+        }
+
+        return strtolower($name) === strtolower($declaringClass->getShortName());
```

Comment:
I don't think this applies anymore: https://3v4l.org/sGLco

Reply:
There's was one revert already :) https://github.com/Roave/BetterReflection/pull/907

https://github.com/Roave/BetterReflection/issues/904#issuecomment-986588835

@ondrejmirtes Is it still relevant?

Reply:
You're right, it last worked in PHP 7. (And it never worked in namespaces.)

Reply:
Let's kill it: we don't support PHP 7.4 anymore.